### ISSUE TITLE
Example doesn't meet idempotency requirement in Worklet spec.

### ISSUE THREAD

**Issue by t1ger-0527** [WICG/animation-worklet#105]
In Animation Worklet example [`animation-with-local-state`](https://github.com/WICG/animation-worklet#animator-migration):

```js
  constructor(options, state) {
    // |options| may be either:
    //  - The user provided options bag passed into the WorkletAnimation constructor on first initialization i.e, {value: 1}.
    //  - The object returned by |destroy| after each migration i.e. {value: 42}.
    this.options_ = options;
    this.state_ = state || {value: Math.random()};
  }
```
The Worklet class constructor initialize `this.state_.value` with `Math.random()` random value.
But according to spec: 
> These specifications must require user agents to always have at least two WorkletGlobalScopes per Worklet and randomly assign a method or set of methods on a class to a particular global scope.

When two of the example worklet got initialized, each of them might get a unique value, which may cause different localTime been set, and animation may jump between two continuous localTime flow (adding 0.1 each step).

In conclusion, I think the code example doesn't meet idempotency requirement in Worklet spec.
Maybe can someone kindly explain the issue for me? Thank you!

**Comment by majido** [comment]
That is correct. 

AnimationWorklet deviates from general Worklet specification in a sense that it allows local state to exist per animator instance. This is not unlike how AudioWorklet also associates an audio node with a [individual processor instance](https://webaudio.github.io/web-audio-api/#instantiation-of-AudioWorkletNode-and-AudioWorkletProcessor) and allows that processor to hold onto local state.

We plan to make the distinction between Stateless and Statefull animator explicit and only allow such behavior in the latter case. See the discussion in issue #87. I will make sure to make this deviation more clear in the specification and other documentation.
